apollo-ast/src/main/kotlin/com/apollographql/apollo3/ast/internal/ExecutableValidationScope.kt

```diff
@@ -507,6 +510,9 @@ internal class ExecutableValidationScope(
   }
 
   private fun fieldsInSetCanMerge(fieldsWithParent: List<FieldWithParent>) {
+    if (fieldOnDisjointTypesMustMerge) {
```

Comment:
I think we're disabling more than what you need here, maybe we could move this a bit deeper? In `sameResponseShapeRecursive` and `haveSameResponseShape`?

Reply:
Yeah, that makes sense. 
I was confused by the issue title - it made me think that we want to disable this whole validation.